extensions/vscode-api-tests/src/singlefolder-tests/terminal.test.ts

```diff
@@ -242,8 +243,31 @@ import { assertNoRpc, poll } from '../utils';
 				}));
 				terminal.sendText('test');
 			});
-			deepStrictEqual(eventState, { isInteractedWith: true });
-			deepStrictEqual(terminal.state, { isInteractedWith: true });
+			strictEqual(eventState.isInteractedWith, true);
```

Comment:
Running into weird problems where shell type may be set but isInteractedWith is suddenly always false???
<img width="784" alt="Screenshot 2025-01-16 at 3 37 12 PM" src="https://github.com/user-attachments/assets/a296aa9d-9996-44cb-b6ef-5393be3311b6" />

Reply:
Maybe onDidChangeTerminalState is somehow being stolen exclusively by shellType event?

Reply:
This looks correct, you should change the test name to mention isInteractedWith and then only assert on `eventState.isInteractedWith`

Reply:
Yeap, I only brought back the { } syntax to better show what is going on - I was expecting isInteractedWith to be true after `terminal.sendText()`

Seems like something fishy is going on, because shellType shows something other than undefined, but isInteractedWith is false..

Reply:
Since the event can fire when shell type changes, it looks like it's firing before it's interacted with if the shell type changed _before_ interaction. So I think we want to do what we do in the shell type test where we check not just the terminal value but that isInteractedWith is true inside the event to make sure it's the one we're interested in.

The fact that shell type may fire doesn't matter for this functionality we're testing, we just want to ignore those ones.